client/src/main.c

```diff
@@ -763,7 +763,7 @@ void clipboardRequest(const LG_ClipboardReplyFn replyFn, void * opaque)
   if (!params.clipboardToLocal)
     return;
 
-  struct CBRequest * cbr = (struct CBRequest *)malloc(sizeof(struct CBRequest()));
+  struct CBRequest * cbr = (struct CBRequest *)malloc(sizeof(*cbr));
```

Comment:
Whoops, clearly an error on my part but I would prefer the brackets to just simply be removed to remain consistent with all the other `malloc` calls through the application.

Reply:
I'm happy to change this, but it doesn't seem to be what the rest of the app does.

```
module/kvmfr.c
183:  kbuf->pages     = kmalloc_array(kbuf->pagecount, sizeof(*kbuf->pages), GFP_KERNEL);

client/src/main.c
118:  SDL_memset(&event, 0, sizeof(event));

repos/LGMP/relacy/test/jtest/jtest.cpp
233:    for (size_t i = 0; i != sizeof(tests)/sizeof(*tests); ++i)
```

Should I still change this to `sizeof *cbr`?

Reply:
No, you missed what I was saying :)

`sizeof(struct CBRequest)`, just remove the invalid `()` on the end. Those other examples you mentioned are also inconsistent, a quick grep will confirm that this here is indeed the style the project uses...

```
client/clipboards/X11/src/x11.c:  this = (struct state *)malloc(sizeof(struct state));
client/clipboards/X11/src/x11.c:  XEvent * s = (XEvent *)malloc(sizeof(XEvent));
client/src/ll.c:  struct ll * list = malloc(sizeof(struct ll));
client/src/ll.c:  struct ll_item * item = malloc(sizeof(struct ll_item));
client/src/main.c:  struct CBRequest * cbr = (struct CBRequest *)malloc(sizeof(struct CBRequest()));
client/src/app.c:  KeybindHandle handle = (KeybindHandle)malloc(sizeof(struct KeybindHandle));
client/fonts/SDL/src/sdl.c:  *opaque = malloc(sizeof(struct Inst));
client/fonts/SDL/src/sdl.c:  LG_FontBitmap * out = malloc(sizeof(LG_FontBitmap));
client/decoders/src/h264.c:  *opaque = malloc(sizeof(struct Inst));
client/decoders/src/null.c:  *opaque = malloc(sizeof(struct Inst));
client/decoders/src/yuv420.c:  *opaque = malloc(sizeof(struct Inst));
client/decoders/src/yuv420.c:  this->pixels = malloc(sizeof(struct Pixel) * (format.width * format.height));
client/parsers/nal.c:  *ptr = (NAL)malloc(sizeof(struct NAL));
client/renderers/OpenGL/opengl.c:  *opaque = malloc(sizeof(struct Inst));
client/renderers/OpenGL/opengl.c:  struct Alert * a = malloc(sizeof(struct Alert));
client/renderers/EGL/fps.c:  *fps = (EGL_FPS *)malloc(sizeof(EGL_FPS));
client/renderers/EGL/cursor.c:  *cursor = (EGL_Cursor *)malloc(sizeof(EGL_Cursor));
client/renderers/EGL/egl.c:  *opaque = malloc(sizeof(struct Inst));
client/renderers/EGL/texture.c:  *texture = (EGL_Texture *)malloc(sizeof(EGL_Texture));
client/renderers/EGL/shader.c:  *this = (EGL_Shader *)malloc(sizeof(EGL_Shader));
client/renderers/EGL/draw.c:  GLfloat * v   = (GLfloat *)malloc(sizeof(GLfloat) * (pts + 1) * 6);
client/renderers/EGL/draw.c:  GLfloat * v   = (GLfloat *)malloc(sizeof(GLfloat) * (pts + 1) * 6);
client/renderers/EGL/splash.c:  *splash = (EGL_Splash *)malloc(sizeof(EGL_Splash));
client/renderers/EGL/alert.c:  *alert = (EGL_Alert *)malloc(sizeof(EGL_Alert));
client/renderers/EGL/model.c:  *model = (EGL_Model *)malloc(sizeof(EGL_Model));
client/renderers/EGL/model.c:  struct FloatList * fl = (struct FloatList *)malloc(sizeof(struct FloatList));
client/renderers/EGL/model.c:  fl->v     = (GLfloat *)malloc(sizeof(GLfloat) * count * 3);
client/renderers/EGL/model.c:  fl->u     = (GLfloat *)malloc(sizeof(GLfloat) * count * 2);
client/renderers/EGL/desktop.c:  *desktop = (EGL_Desktop *)malloc(sizeof(EGL_Desktop));
common/src/stringlist.c:  StringList sl = malloc(sizeof(struct StringList));
common/src/stringlist.c:  sl->list         = malloc(sizeof(char *) * sl->size);
common/src/platform/windows/thread.c:  *handle             = (LGThread *)malloc(sizeof(LGThread));
common/src/platform/windows/event.c:  LGEvent * event = (LGEvent *)malloc(sizeof(LGEvent));
common/src/platform/windows/event.c:  LGEvent * event = (LGEvent *)malloc(sizeof(LGEvent));
common/src/platform/windows/event.c:  HANDLE * handles = (HANDLE *)malloc(sizeof(HANDLE) * count);
common/src/platform/windows/time.c:  LGTimer * ret = malloc(sizeof(LGTimer));
common/src/platform/windows/ivshmem.c:    (struct IVSHMEMInfo *)malloc(sizeof(struct IVSHMEMInfo));
common/src/platform/linux/thread.c:  *handle = (LGThread*)malloc(sizeof(LGThread));
common/src/platform/linux/time.c:  LGTimer * ret = malloc(sizeof(LGTimer));
common/src/platform/linux/ivshmem.c:    (struct IVSHMEMInfo *)malloc(sizeof(struct IVSHMEMInfo));
common/src/option.c:    state.options[state.oCount + i] = (struct Option *)malloc(sizeof(struct Option));
common/src/option.c:      group->options    = malloc(sizeof(struct Option *));

```

Reply:
Ah, got it.

The `sizeof(var)` pattern is a bit less error-prone and avoids the class of bugs where the wrong type is used on the right-hand side and the allocation is of the wrong size (like here), but I've changed it to repeat the type for consistency.

Reply:
Thanks, I would accept a PR that switches the project over to using this style as the reasoning is sound, but until the standard is changed, yeah, keep it consistent :+1: